src/crypto-providers/util.ts

```diff
@@ -569,6 +595,30 @@ const validateVotingRegistrationAddressType = (addressType: number) => {
   }
 }
 
+const certificatesContainScriptHashStakeCredentials = (certificates: _Certificate[]) => (
+  (certificates.filter((cert) => cert.type in [
+    TxCertificateKeys.DELEGATION,
+    TxCertificateKeys.STAKING_KEY_REGISTRATION,
+    TxCertificateKeys.STAKING_KEY_DEREGISTRATION,
+  ]) as (
+      _DelegationCert |
+      _StakingKeyRegistrationCert |
+      _StakingKeyDeregistrationCert
+    )[]).some(
+    (cert) => cert.stakeCredentials.type === StakeCredentialsKeys.SCRIPT_HASH,
+  )
+)
+
+const withdrawalsContainScriptHashStakeCredentials = (withdrawals: _Withdrawal[]) => withdrawals.some(
+  (withdrawal) => withdrawal.stakeCredential.type === StakeCredentialsKeys.SCRIPT_HASH,
+)
+
+const hasSomeScriptHashStakeCredentials = (unsignedTxParsed: _UnsignedTxParsed) => (
```

Comment:
The name of this is misleading. Let's remove the `unsignedTxParsed.scriptWitnesses.length > 0` part and only use it in `determineSigningMode`. Also, I presume the script witnesses we are talking about are for native scripts only; that should be explicitly clear from just the field name.

Reply:
I removed it, and now instead of calling `hasSomeScriptHashStakeCredentials` the `isMultisigTransaction` function is exported. Hopefully that's better.